libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -589,29 +601,49 @@ - (BOOL)hasCommunityChanged {
     }
 }
 
-- (void)userChanged:(SFUserAccountChange)change {
-    if (![self.lastChangedOrgId isEqualToString:self.currentUser.credentials.organizationId]) {
-        self.lastChangedOrgId = self.currentUser.credentials.organizationId;
+- (void)userChanged:(SFUserAccount *)user change:(SFUserAccountChange)change {
+    [self notifyUserChange:SFUserAccountManagerDidChangeUserDataNotification withUser:user andChange:change];
+}
+
+- (void)notifyCurrentUserChange:(SFUserAccountChange)change {
```

Comment:
Ah.  This feels like an undesirable mashing up of two different notifications.  I'm wondering if we should just keep two notifications—they really fulfill two different use cases.

One is listening for the fact that the current user has switched, which is really about changing to an entirely different entity, so granular parsing of what changed below that is really beside the point.  It's a totally different user.  It's all presumably changed, at that point.

The other is listening for specific changes on a given user, where granular denoting of what changed is desirable.

They're really two different things.  Mashing them together distorts the purpose of the notification.